src/Command/QueueCommand.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * Queue command.
+ *
+ * For listening to SQS.
+ */
+
+namespace eLife\Bus\Command;
+
+use eLife\ApiClient\Exception\BadResponse;
+use eLife\Bus\Queue\QueueItem;
+use eLife\Bus\Queue\QueueItemTransformer;
+use eLife\Bus\Queue\WatchableQueue;
+use eLife\Logging\Monitoring;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Throwable;
+
+abstract class QueueCommand extends Command
+{
+    protected $logger;
+    protected $queue;
+    protected $transformer;
+    protected $limit;
+    protected $monitoring;
+    protected $topic;
+
+    public function __construct(string $topic, LoggerInterface $logger, WatchableQueue $queue, QueueItemTransformer $transformer, Monitoring $monitoring, callable $limit)
```

Comment:
should not be a `$topic` but a `$queue`. But actually we already pass in `$queue` so if the only thing this is doing with `string $topic` is to log its its name, I would remove it. If we really need it for logging we can add `__toString()` to `WatchableQueue`